really-core/src/main/scala/io/really/gorilla/Replayer.scala

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (C) 2014-2015 Really Inc. <http://really.io>
+ */
+
+package io.really.gorilla
+
+import akka.actor._
+import akka.contrib.pattern.DistributedPubSubMediator.Unsubscribe
+import akka.util.Timeout
+import io.really.gorilla.Replayer.SnapshotResult._
+import io.really.gorilla.Replayer.{ EndOfEventsLog, SnapshotResult }
+import io.really.model.CollectionActor.{ GetState, State }
+import io.really.{ R, Revision, ReallyGlobals }
+import akka.pattern.{ AskTimeoutException, ask, pipe }
+import play.api.libs.json.JsObject
+import io.really.CommandError.ObjectNotFound
+import io.really.protocol.SubscriptionFailure
+
+import scala.concurrent.Future
+import scala.slick.lifted.TableQuery
+import scala.slick.driver.H2Driver.simple._
+import scala.util.control.NonFatal
+import scala.concurrent.ExecutionContext.Implicits.global
+import EventLogs._
+
+class Replayer(globals: ReallyGlobals, objectSubscriber: ActorRef, rSubscription: RSubscription,
+    maxMarker: Option[Revision])(implicit session: Session) extends Actor with ActorLogging with Stash {
+
+  val replayerId = s"Replayer ${rSubscription.pushChannel.path}$$${rSubscription.r}"
+
+  val min = rSubscription.rev
+  private[gorilla] val r = rSubscription.r
+
+  context.watch(objectSubscriber)
+  //TODO Watch Gorilla to notify the ObjectSubscriber that the subscription failed in case of Gorilla's termination
+
+  maxMarker.map {
+    max: Revision =>
+      if (min == max) {
+        context.become(servePushUpdates)
+      } else if (min > max && (min - max) <= globals.config.GorillaConfig.advancedRevisionLimit) {
+        context.become(waitingCorrectRevision)
+      } else if (min > max) {
+        shutdownWithObjectSubscriptionAck("Subscription revision advances the current known revision with more than" +
```

Comment:
You should do this check in gorilla center before replying on the subscription manager that the subscription succeed.